File: src/main/java/com/flexpoker/table/command/repository/TableEventRepository.java
package com.flexpoker.table.command.repository;

import java.util.List;
import java.util.UUID;

import com.flexpoker.table.command.events.TableEvent;

public interface TableEventRepository {

    List<TableEvent> fetchAll(UUID id);

    List<TableEvent> setEventVersionsAndSave(int basedOnVersion, List<TableEvent> events);

}

File: src/main/kotlin/com/flexpoker/table/command/repository/TableEventRepository.kt
package com.flexpoker.table.command.repository

import com.flexpoker.table.command.events.TableEvent
import java.util.UUID

interface TableEventRepository {
    fun fetchAll(id: UUID): List<TableEvent>
    fun setEventVersionsAndSave(basedOnVersion: Int, events: List<TableEvent>): List<TableEvent>
}